### ISSUE TITLE
Package request: clickhouse

### ISSUE THREAD

**Issue by sirikid** [void-linux/void-packages#12563]
ClickHouse is a free analytic DBMS for big data. 
https://github.com/yandex/ClickHouse
https://clickhouse.yandex/

**Comment by knusbaum** [comment]
Trying to figure out packaging: https://github.com/yandex/ClickHouse/issues/6128

**Comment by Johnnynator** [comment]
@knusbaum how many if these gitmodules are actually needed? Most of these just look like libs that we have already in the repo. (`-DUNBUNDLED=ON` lets ClickHouse use system libs)

**Comment by knusbaum** [comment]
@Johnnynator Yes, I tried doing that. I was somewhat successful with that process, but ran into things that are ClickHouse specific, for example: https://github.com/ClickHouse-Extras/ssl, which is a necessary dependency, but has no releases and is not bundled with the source tarball.

Also, when I discussed using system libs with ClickHouse, I got this response:
```
Don't do it. With libraries from OS packages ClickHouse will work in a slightly different, possibly wrong way. This is intended only as a "demo" version for official Debian/Ubuntu repo.
```
I find that response odd, but if they want to package all the dependencies or at least provide hard version numbers and checksums, or really anything that can give us a reproducible build, then I guess that can work. 

Regardless I think they need to provide *something* for ClickHouse-Extras/ssl and others like it. Checking out master on build doesn't sound like a good idea.
